### DISCUSSION TITLE
HTTPS issues (without custom domain)

### DISCUSSION THREAD

**Discussion by Conengmo** [community/community#65916] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Multiple users reported issues reaching our projects documentation on Pages: https://python-visualization.github.io/folium/.

The issues seem related to HTTPS/SSL, for example they report the error message `ERR_SSL_PROTOCOL_ERROR` in Chrome.

I can't reproduce the issue myself unfortunately. We **don't** use a custom domain, makes it hard to search for this issue since everything I read about this is from users who do have a custom domain.

See https://github.com/python-visualization/folium/issues/1795 for the related issue.

- What could be causing this? 
- Is there something we can do? 
- Or what can we tell users who have this issue what they should do?

Any help is appreciated! I'm curious to learn more about this.

**Comment by GabrieleMajocchi** [comment]
The "ERR_SSL_PROTOCOL_ERROR" error in Chrome typically indicates an issue with the SSL/TLS handshake when establishing a secure connection to a website. Since you don't have a custom domain for your GitHub Pages site (i.e., it's hosted at https://python-visualization.github.io/folium/), the SSL/TLS configuration is managed by GitHub itself. Here are some possible reasons for this error and steps to address it:

Network Issues: Some users might experience network issues that prevent the SSL/TLS handshake from completing successfully. They can try the following steps:

Clear their browser cache and cookies.
Restart their router or modem.
Check if they are behind a firewall or using a VPN that might interfere with the SSL handshake.
Browser Issues: Occasionally, browser extensions or configurations can cause SSL/TLS errors. Users can try:

Temporarily disabling browser extensions to see if one of them is causing the problem.
Trying a different web browser to see if the issue persists.
Outdated Browsers: Encourage users to ensure they are using the latest version of their web browser, as older versions may have compatibility issues with certain websites.

GitHub Pages Status: Sometimes, GitHub Pages itself may experience downtime or issues with SSL/TLS. You can check the status of GitHub services at [GitHub Status](https://www.githubstatus.com/). If GitHub Pages is experiencing issues, it's a server-side problem that users cannot fix on their end.

Mixed Content: Ensure that all resources (e.g., images, scripts, stylesheets) on your GitHub Pages site are served over HTTPS. Mixed content (a combination of HTTP and HTTPS resources) can cause SSL/TLS errors. Use relative URLs or make sure all resources are loaded securely.

TLS Compatibility: Check if your GitHub Pages site is configured to use modern TLS versions and encryption ciphers. GitHub typically maintains secure configurations, but it's worth confirming.

Caching: Some users might be experiencing cached versions of your site that had SSL/TLS issues. They can try accessing the site in an incognito or private browsing window to bypass their browser cache.

ISP or DNS Issues: Rarely, the user's Internet Service Provider (ISP) or DNS resolver could be causing problems. Users can try changing their DNS resolver to a trusted public DNS service like Google DNS or Cloudflare DNS to see if it resolves the issue.

GitHub Repository Settings: Double-check your repository settings on GitHub to ensure that it's set to use GitHub Pages with HTTPS. To do this, go to the repository's "Settings" > "Pages" and make sure it's configured correctly.

GitHub Issue: Monitor the GitHub issue you mentioned (python-visualization/folium#1795) for any updates or resolutions from other users who may have experienced the same problem.

If users continue to experience issues after trying the above steps, they should provide detailed information about their setup, including their browser version, operating system, and any error messages, so you can better assist them in identifying the cause of the problem. Additionally, if you notice a pattern of issues among multiple users, you can consider reaching out to GitHub Support for further assistance.

**Comment by Conengmo** [comment]
Pretty sure that's an answer generated by a chatbot :p Still useful I suppose.
